UnityProject/Assets/Scripts/Objects/Singularity.cs

```diff
@@ -232,9 +233,18 @@ private void SingularityUpdate()
 			dynamicScale = GetDynamicScale();
 
 			//Radiation Pulse
-			var strength = Mathf.Max(((float) CurrentStage + 1) / 6 * maxRadiation, 0);
+			var radStrength = Mathf.Max(((float) CurrentStage + 1) / 6 * maxRadiation, 0);
 
-			RadiationManager.Instance.RequestPulse(registerTile.WorldPositionServer, strength, objectId);
+			RadiationManager.Instance.RequestPulse(registerTile.WorldPositionServer, radStrength, objectId);
+
+			if(UnityEngine.Random.Range(0,15) == 0)
+            {
+				int EMPStrength = UnityEngine.Random.Range((int)CurrentStage * 100, (int)CurrentStage * 100 + 200);
+				Vector3Int EMPPosition = registerTile.WorldPositionServer;
+				EMPPosition.x += UnityEngine.Random.Range(-3 - (int)CurrentStage, 3 + (int)CurrentStage);
+				EMPPosition.y += UnityEngine.Random.Range(-3 - (int)CurrentStage, 3 + (int)CurrentStage);
```

Comment:
What does this curve look like? How did you find this to be the formula?

Reply:
There's no formula here, just made this to offset pulse position from singulo center